testutil/integration/router.go

```diff
@@ -88,16 +87,17 @@ func NewIntegrationApp(
 	grpcRouter.SetInterfaceRegistry(interfaceRegistry)
 	bApp.SetGRPCQueryRouter(grpcRouter)
 
-	if keys[consensusparamtypes.StoreKey] != nil {
+	if keys[consensus] != nil {
 		// set baseApp param store
-		consensusParamsKeeper := consensusparamkeeper.NewKeeper(appCodec, runtime.NewEnvironment(runtime.NewKVStoreService(keys[consensusparamtypes.StoreKey]), log.NewNopLogger(), runtime.EnvWithQueryRouterService(grpcRouter), runtime.EnvWithMsgRouterService(msgRouter)), authtypes.NewModuleAddress("gov").String())
-		bApp.SetParamStore(consensusParamsKeeper.ParamsStore)
-		consensusparamtypes.RegisterQueryServer(grpcRouter, consensusParamsKeeper)
-
-		params := cmttypes.ConsensusParamsFromProto(*simtestutil.DefaultConsensusParams) // This fills up missing param sections
-		err := consensusParamsKeeper.ParamsStore.Set(sdkCtx, params.ToProto())
-		if err != nil {
-			panic(fmt.Errorf("failed to set consensus params: %w", err))
+		consensusParamsKeeper, ok := modules[consensus].(baseapp.ParamStore)
```

Comment:
I don't think this is correct, we need to have the collection embedded in the consensus params keeper for this to work.

Reply:
Additonally, modules is an AppModule, so we need to get the keeper from the appmodule. So that will leak its type unfortunately :(

Reply:
need to check how to make this work without the consensus keeper being imported